src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/1.sql

```diff
@@ -273,6 +273,392 @@ ON dbo.CompartmentAssignment
 WHERE IsHistory = 0
 WITH (DATA_COMPRESSION = PAGE)
 
+GO
+
+/*************************************************************
+    Reference Search Param
+**************************************************************/
+
+CREATE TYPE dbo.ReferenceSearchParamTableType_1 AS TABLE  
+(
+    SearchParamId smallint NOT NULL,
+    BaseUri varchar(128) NULL,
+    ReferenceResourceTypeId smallint NOT NULL,
+    ReferenceResourceId varchar(64) NOT NULL,
+    ReferenceResourceVersion int NULL
+)
+
+CREATE TABLE dbo.ReferenceSearchParam
+(
+    ResourceSurrogateId bigint NOT NULL,
+    SearchParamId smallint NOT NULL,
+    BaseUri varchar(128) NULL,
+    ReferenceResourceTypeId smallint NOT NULL,
+    ReferenceResourceId varchar(64) NOT NULL,
```

Comment:
I am not sure if 64 is long enough. In the case of external resource reference, this will be an URL.